zospy/functions/lde.py

```diff
@@ -71,3 +71,67 @@ def surface_change_type(surface: _ZOSAPI.Editors.LDE.ILDERow, new_type: constant
     # Apply
     new_surface_type_settings = surface.GetSurfaceTypeSettings(new_type)
     surface.ChangeType(new_surface_type_settings)
+
+
+def find_surface_by_comment(
+    lde: _ZOSAPI.Editors.LDE, comment: str, case_sensitive: bool = False
+) -> list[_ZOSAPI.Editors.LDE.ILDERow]:
+    """Returns a list of surfaces from the LDE that have the supplied string as Comment.
+
+    In case of multiple matches, the surfaces are returned in ascending order.
+
+    Parameters
+    ----------
+    lde: ZOSAPI.Editors.LDE
+        The Lens Data Editor (LDE)
+    comment: str
+        String that is searched for in the Comment column of the NCE.
+    case_sensitive: bool=False
+        Flag that specifies whether the search is case-sensitive or not. Defaults to False.
+
+    Returns
+    -------
+    list[_ZOSAPI.Editors.LDE.ILDERow]
```

Comment:
Please do not use the underscore when specifying ZOSAPI types in docstrings (the underscore only has a meaning in the context of "importing" the stub files). This also goes for the NCE function.

Reply:
Updated in the LDE function!

Reply:
Done in the NCE one as well